spark-bigquery-dsv2/spark-3.1-bigquery/src/main/java/com/google/cloud/spark/bigquery/v2/BigQueryTableProvider.java

```diff
@@ -15,33 +15,88 @@
  */
 package com.google.cloud.spark.bigquery.v2;
 
+import static scala.collection.JavaConversions.mapAsJavaMap;
+
+import com.google.cloud.bigquery.connector.common.BigQueryConfigurationUtil;
+import com.google.cloud.bigquery.connector.common.BigQueryConnectorException;
+import com.google.common.collect.ImmutableMap;
 import com.google.inject.Injector;
 import java.util.Map;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.catalyst.analysis.NoSuchTableException;
+import org.apache.spark.sql.connector.catalog.Identifier;
+import org.apache.spark.sql.connector.catalog.SupportsCatalogOptions;
 import org.apache.spark.sql.connector.catalog.Table;
 import org.apache.spark.sql.connector.catalog.TableProvider;
 import org.apache.spark.sql.connector.expressions.Transform;
 import org.apache.spark.sql.types.StructType;
 import org.apache.spark.sql.util.CaseInsensitiveStringMap;
 
-public class BigQueryTableProvider extends BaseBigQuerySource implements TableProvider {
+public class BigQueryTableProvider extends BaseBigQuerySource
+    implements TableProvider, SupportsCatalogOptions {
 
+  private static final String DEFAULT_CATALOG_NAME = "bigquery";
+  private static final String DEFAULT_CATALOG = "spark.sql.catalog." + DEFAULT_CATALOG_NAME;
   private static final Transform[] EMPTY_TRANSFORM_ARRAY = {};
 
   @Override
   public StructType inferSchema(CaseInsensitiveStringMap options) {
-    return getTable(null, EMPTY_TRANSFORM_ARRAY, options).schema();
+    return getBigQueryTableInternal(options).schema();
   }
 
   @Override
   public Table getTable(
       StructType schema, Transform[] partitioning, Map<String, String> properties) {
-    Injector injector = InjectorFactory.createInjector(schema, properties);
-    BigQueryTable table = new BigQueryTable(injector, schema);
-    return table;
+    setupDefaultSparkCatalog(SparkSession.active());
+    return getBigQueryTableInternal(schema, properties);
+  }
+
+  private BigQueryTable getBigQueryTableInternal(Map<String, String> properties) {
+    return getBigQueryTableInternal(null, properties);
+  }
+
+  private BigQueryTable getBigQueryTableInternal(
```

Comment:
Can we rename this to `getInternalBigQueryTable` or just `getBigQueryTable`?

Reply:
It is a common patten, as the the method is internal, not the BigQueryTable